bundles/org.openhab.core.voice/src/main/java/org/eclipse/smarthome/core/voice/text/ExpressionCardinality.java

```diff
@@ -62,7 +62,7 @@ ASTNode parse(ResourceBundle language, TokenList tokenList) {
             node.setChildren(nodes.toArray(new ASTNode[0]));
             node.setRemainingTokens(list);
             node.setSuccess(true);
-            node.setValue(atMostOne ? (values.size() > 0 ? values.get(0) : null) : values.toArray());
+            node.setValue(atMostOne ? (!values.isEmpty() ? values.get(0) : null) : values.toArray());
```

Comment:
This tenary operator can be inverted:
```suggestion
            node.setValue(atMostOne ? (values.isEmpty() ? null : values.get(0)) : values.toArray());
```

Reply:
I also prefer getting rid of the exclamation marks where possible because they are easily overlooked (https://github.com/openhab/openhab-core/issues/1022#issuecomment-538877251) :exclamation: